src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -1312,10 +1312,17 @@ Status ServiceBasedErrorInfoAccessor::AsyncReportJobError(
     const StatusCallback &callback) {
   auto job_id = JobID::FromBinary(data_ptr->job_id());
   RAY_LOG(DEBUG) << "Publishing job error, job id = " << job_id;
-  Status status = client_impl_->GetGcsPubSub().Publish(
-      ERROR_INFO_CHANNEL, job_id.Hex(), data_ptr->SerializeAsString(), callback);
-  RAY_LOG(DEBUG) << "Finished publishing job error, job id = " << job_id;
-  return status;
+  rpc::ReportJobErrorRequest request;
```

Comment:
I have one question. How does the job error look like? Who was the subscriber, and are subscribers still getting data?

Reply:
We barely publish job error data, and only does dashboard subscribe it for showing what happened to a failed job.
Before publishers publish job error msg to redis directly and subscribers receive that. After this change they send this message to gcs first and let gcs publish this message. It is  transparent to subscribers' side.

Reply:
One usage is that if you start a Python driver in terminal, you can see the error messages in the driver output even though errors are generated in another process or even on another node.